src/python/pants/backend/python/macros/python_requirements_test.py

```diff
@@ -104,6 +105,55 @@ def test_requirements_txt(rule_runner: RuleRunner) -> None:
     )
 
 
+def test_multiple_versions(rule_runner: RuleRunner) -> None:
+    """This tests that we correctly create a new python_requirement_library for each unique
+    dependency name in a requirements.txt file, grouping duplicated dependency names to handle
+    multiple requirement strings per PEP 508."""
+
+    assert_python_requirements(
+        rule_runner,
+        "python_requirements(module_mapping={'ansicolors': ['colors']})",
+        dedent(
+            """\
+            Django>=3.2
+            Django==3.2.7
+            confusedmonkey==86
+            repletewateringcan>=7
```

Comment:
You can leave these out if you want. We already test that we can handle requirements like that - we only need to check that the `Django` edge case works.

Reply:
The intention for adding further dependencies was that we verify that we correctly handle a case where there's multiple constraints for one package name, but only one constraint for others. Do you feel strongly about not needing that?

Reply:
Nope, totally fine to still have it.